Analysis/interface/GenLepton.h

```diff
@@ -243,8 +243,15 @@ class GenLepton {
             size_t last_mother_index = NoneIndex;
 
             if(!particle->motherRefVector().empty()) {
-                for(const auto& mother : particle->motherRefVector())
+		int Nmother = 0;
```

Comment:
I think it is better to fix the limitation at the level of filling TauTuple and keep `GenLepton.h` logic intact.
https://github.com/cms-tau-pog/TauMLTools/blob/master/Production/plugins/TauTupleProducer.cc#L401-L420